src/db.c

```diff
@@ -4128,206 +4155,1718 @@ db_speaker_clear_all(void)
+  queue_item->data_kind = sqlite3_column_int(queue_enum->stmt, 4);
+  queue_item->media_kind = sqlite3_column_int(queue_enum->stmt, 5);
+  queue_item->song_length = (uint32_t)sqlite3_column_int(queue_enum->stmt, 6);
+  queue_item->path = strdup_if((char *)sqlite3_column_text(queue_enum->stmt, 7), keep_item);
+  queue_item->virtual_path = strdup_if((char *)sqlite3_column_text(queue_enum->stmt, 8), keep_item);
+  queue_item->title = strdup_if((char *)sqlite3_column_text(queue_enum->stmt, 9), keep_item);
+  queue_item->artist = strdup_if((char *)sqlite3_column_text(queue_enum->stmt, 10), keep_item);
+  queue_item->album_artist = strdup_if((char *)sqlite3_column_text(queue_enum->stmt, 11), keep_item);
+  queue_item->album = strdup_if((char *)sqlite3_column_text(queue_enum->stmt, 12), keep_item);
+  queue_item->genre = strdup_if((char *)sqlite3_column_text(queue_enum->stmt, 13), keep_item);
+  queue_item->songalbumid = sqlite3_column_int64(queue_enum->stmt, 14);
+  queue_item->time_modified = sqlite3_column_int(queue_enum->stmt, 15);
+  queue_item->artist_sort = strdup_if((char *)sqlite3_column_text(queue_enum->stmt, 16), keep_item);
+  queue_item->album_sort = strdup_if((char *)sqlite3_column_text(queue_enum->stmt, 17), keep_item);
+  queue_item->album_artist_sort = strdup_if((char *)sqlite3_column_text(queue_enum->stmt, 18), keep_item);
+  queue_item->year = sqlite3_column_int(queue_enum->stmt, 19);
+  queue_item->track = sqlite3_column_int(queue_enum->stmt, 20);
+  queue_item->disc = sqlite3_column_int(queue_enum->stmt, 21);
+
+  return 0;
+}
+
+int
+db_queue_enum_start(struct db_queue_enum *queue_enum)
+{
+  int ret;
+
+  db_transaction_begin();
+
+  ret = queue_enum_start(queue_enum);
+
+  if (ret < 0)
+    db_transaction_rollback();
+
+  return ret;
+}
+
+void
+db_queue_enum_end(struct db_queue_enum *queue_enum)
+{
+  if (!queue_enum->stmt)
+    return;
+
+  queue_enum_end(queue_enum);
+  db_transaction_end();
+}
+
+int
+db_queue_enum_fetch(struct db_queue_enum *queue_enum, struct db_queue_item *queue_item)
+{
+  return queue_enum_fetch(queue_enum, queue_item, 0);
+}
+
+int
+db_queue_get_pos(uint32_t item_id, char shuffle)
+{
+#define Q_TMPL "SELECT pos FROM queue WHERE id = %d;"
+#define Q_TMPL_SHUFFLE "SELECT shuffle_pos FROM queue WHERE id = %d;"
+
+  char *query;
+  int pos;
+
+  if (shuffle)
+    query = sqlite3_mprintf(Q_TMPL_SHUFFLE, item_id);
+  else
+    query = sqlite3_mprintf(Q_TMPL, item_id);
+
+  pos = db_get_count(query);
+
+  sqlite3_free(query);
+
+  return pos;
+
+#undef Q_TMPL
+#undef Q_TMPL_SHUFFLE
+}
+
+int
+db_queue_get_pos_byfileid(uint32_t file_id, char shuffle)
+{
+#define Q_TMPL "SELECT pos FROM queue WHERE file_id = %d LIMIT 1;"
+#define Q_TMPL_SHUFFLE "SELECT shuffle_pos FROM queue WHERE file_id = %d LIMIT 1;"
+
+  char *query;
+  int pos;
+
+  if (shuffle)
+    query = sqlite3_mprintf(Q_TMPL_SHUFFLE, file_id);
+  else
+    query = sqlite3_mprintf(Q_TMPL, file_id);
+
+  pos = db_get_count(query);
```

Comment:
Should i do this in this pr or in a separate one (this will change otherwise unrelated code)?